public/js/form_helpers.js

```diff
@@ -20,42 +20,64 @@ $(document).ready(function(){
         .on('focusout', function() {
             validation($(this));
         });
+    $('.mask-zip')
+        .on('focusout', function() {
+            validateZip($(this));
+        })
 });
 
 function initMask(){
     $('.mask-zip').mask('00000');
-    $('.mask-phone').mask('(000) 000-0000');
+    $('.mask-phone').mask('0000000000');
 }
 
+// Javascript validation for Referral form.
 function submitAll(){
     var fname = document.getElementById("pers_firstname");
     var lname = document.getElementById("pers_lastname");
-    var refEmail = document.getElementById("ref_email");
+    var pers_zip = document.getElementById("pers_zip");
+    var ref_email = document.getElementById("ref_email");
     var reqForm = document.getElementById("participant_info");
-
-    if (reqForm.checkValidity() == false) {
-        document.getElementById("pers_title").focus();
-        reqForm.classList.add("was-validated");
-        if(lname.value.length === 0)
-            lname.focus();
-        if(fname.value.length === 0 )
+    var card = document.getElementById("pers_title");
+    var card2 = document.getElementById("pers_referring_party_info");
+    // Handles all validation when the user hits the submit button.
+    if (reqForm.checkValidity() === false) {
+        if (fname.value.length === 0 ) {
+            card.focus();
             fname.focus();
+        } else if (lname.value.length === 0) {
+            card.focus();
+            lname.focus();
+        } else {
+            $('#ref_email').addClass("is-invalid");
+            card2.focus();
+            ref_email.focus();
+        }
+    } else if (pers_zip.value.length !== 5 && pers_zip.value.length !== 0) {
```

Comment:
You already wrote a `validateZip` function, you should change this to `!validZip(pers_zip.value)` . I do see that the `validateZip` function you currently have is also responsible for adding and removing a class to an element, you might write a `validZip` function that checks the length and revise `validateZip` to call` validZip` and then take care of adding/removing classes from the element sent.

Reply:
Done